### DISCUSSION TITLE
Silmarils games sound issue.

### DISCUSSION THREAD

**Discussion by AnonTet** [dosbox-staging/dosbox-staging#3467] [General]
It came to my attention on eXoDOS discord that Silmarils games have sound issues that don't happen with dosbox-x. Namely the "Ding" sound at the end of "silmarils" logo show when the game is launched.

Tested with Boston Bomb Club but in Discord the issue was noticed on Bunny Bricks. Unfortunately I don't remember if Boston bomb club had any music to test more but the "ding" is definitely not there.

Is this a config issue or an improvement possibility of -staging? 

PS: Searching for "Silmarils" yield no results.

**Comment by madbrain76** [comment]
I believe Boston Bomb Club only supports only the PC speaker in the DOS version, as did all Silmarils games at that time.

That changed after the summer of 1992, when I wrote the code for soundcards on PC for Silmarils. It was first released starting with Ishar. I was just 16 at the time., on summer break and internship I no longer have the source, though. I believe all subsequent Silmarils games for PC used this code. I also wrote the setup program, in assembly as well. I just downloaded Bunny Bricks, and see my name at the top of the setup. I never played this game before, or any subsequent Silmarils game, as I moved to the US a few years later.

All the audio in Ishar was PCM, actually mixed in real-time, like a MOD file on Amigas at the time, and the demo scene on PCs. I do remember my code for both the audio uses tricks to get the 1-bit PC speaker sound to actually be somewhat tolerable, using a very high interrupt rate. It is done similarly for Adlib, except it produces 6-bit audio. On Sound Blaster, it is much different. It uses DMA / interrupt to pass audio to the soundcard. One tricky issue was getting the interrupt and DMA channels right. I wrote some code to auto-detect it. Perhaps it is not 100% fool-proof in an emulator context. As I recall, the "BLASTER" environment variable, if set, takes precedence over the auto-detection. Reading the specs on that environment variable, the Tx component is the soundcard type/revision. . I checked the release date of the Sound Blaster 16. It is June of 1992. I wrote the SB code for Silmarils in July, while on school break. It's extremely unlikely I would have had an SB16 on hand at that time to test with.  The "Hx" parameter for "high DMA" of the environment variable was introduced on the SB16. All previous SBs were 8-bit card and couldn't use DMA 4-6. It's quite possible that my code cannot parse the environment variable because of that. You can try removing the H5 from the variable, or unsetting the variable altogether and trying SB mode again.

Edit: I just tried it. Unsetting the BLASTER environment variable fixes the music on the title page. It doesn't sound half-bad for 8-bit audio of that era.! My auto-detection code worked, as setup automatically selected Sound Blaster. At least it did in regular DOSBox. I haven't installed the Staging fork yet. Sadly, the game is stuck while that music plays in Sound Blaster mode, and unable to be started.

I tried the Adlib version. It sounds positively awful on the initial screens. But the game can actually start with the Enter key. After that, there is a tune playing on the OPL2 chip, not PCM anymore, with the hack that I implemented for AdLib. This AdLib version seems to be the only one in which the game can be started. I can't even start the game properly with the "No sound" option. I take no responsibility for these bugs :-) Not sure if the game started issue is emulator-related or not. It probably worked fine on bare metal at the time, with <33 MHz CPUs and 512-640KB of RAM. I'm going to check out DOSBox Staging to see if it's any better than the regular version. And try Ishar (1) as well, as it uses my audio code throughout both for music and sound effects.

**Reply by interloper98** [reply]
I've got an incoming PR (https://github.com/dosbox-staging/dosbox-staging/pull/3772) to fix these games.  Here's some details:

These are OPL-capable, but not SB PCM/DMA capable:
- Boston Bomb Club (1991)
- Crystals of Arborea (1990)
- Metal Mutant (1991)
- Storm Master (1992)

Use this configuration for them:

```ini
[sblaster]
sbtype = none
oplmode = opl2

[autoexec]
@echo off
set blaster=
mount c .
c:
mixer opl 300 r15 c25
mousectl -s 200 200
autotype f2
boston.exe / metal / crystal / storm
```

These are SB PCM/DMA & OPL-capable:
- Bunny Bricks (1993)
- Transantartica (1993)
- Ishar 2 (1993)
- Ishar 1 (1992)

Use this configuration for them (and configure them for Sound Blaster):

```ini
[sblaster]
sbtype = sb2

[autoexec]
@echo off
set blaster=
mount c .
c:
mixer opl 200 r15 c25 sb 200
mousectl -s 200 200
start.exe
```

**Comment by johnnovak** [comment]
@madbrain76 Welcome, it's super cool you found us and shared the above information with us 😎 🤘🏻 

We already have a ticket for this one:
https://github.com/dosbox-staging/dosbox-staging/issues/3504

PCM playback tricks via PC Speaker and AdLib are fully supported in Staging. But yeah, our problem is mainly the missing sound when the logo gets displayed in the intro. Looking forward to your further informative comments on the matter! 😄

**Comment by madbrain76** [comment]
@johnnovak ,
I believe the initial sound with the Silmarils logo was already in games prior to Ishar, and thus must have been handled by different code, and would sound only on PC speaker. Not certain what was done for that initial sound for Ishar and subsequent game. I assume they didn't leave it on PC speaker and play the rest on the soundcard - that would be strange. Presumably some log could tell which audio device is active while that sound is being played back.

Time to install DOSbox staging and reinstall DOSBos-X also. It's unfortunate that no single build can properly play all games.

**Reply by johnnovak** [reply]
> It's unfortunate that no single build can properly play all games.

Well, we're aiming to be able to do that 😏  It's just a matter of time 😄

**Comment by madbrain76** [comment]
Yes, I know. Everybody else does, too. The feature set differs between forks, though.. One of my favorite ones is the ability to record MIDI streams from games. I grew up with an LAPC-1, not just a Soundblaster. I no longer own, but I like Munt. And I have external MIDI hardware as well - Fantom XR, which can be used for GM games. Apparently not for GS, though.
As far as I can tell, this feature isn't in DosBox staging.
Anyway, I have a whole bunch of bugs I just found in MuseScore that I need to report, before I get to installing Dosbox staging. But I'll do it for sure.

**Reply by johnnovak** [reply]
> As far as I can tell, this feature isn't in DosBox staging.

You can output DOS MIDI to any MIDI port on your system in Staging, just like in original DOSBox. Look into the `midiconfig` and `mididevice` settings.

**Reply by madbrain76** [reply]
Thanks. That's great to hear. I actually read the Wiki docs for DosBox-X today and could not find any mention of the possibility of using native Windows devices with midiconfig and mididevice . Î hope that doesn't mean the feature is an afterthought.
The possibility of recording DOS app MIDI output to MIDI files also is not mentioned on the Wiki. To me, it's a very important feature. I have tried many sequencers , both hardware and software, and I never found one that could record the proper MIDI stream inclusive of all necessary Sysex messages. I found a ticket on DOSBos-X that calls this a niche feature. Certainly not to me.
Anyway, it's a bit strange that I RTFM before installing the software, but I'm getting old. And I usually use docs to decide whether I want to use software or not. And without your comment, I wouldn't have known these features were available and not stripped in the fork.

**Reply by johnnovak** [reply]
A comprehensive user manual is in the works (done mostly by me), but my time and energy is unfortunately limited, so it's going slowly. All this will be documented in detail. Yeah, the wiki is a volunteer-based effort, and it's not the most consistent, overall, but a very good source of information on many topics.

I'm also a hobby musician, and I'm super into all sorts of audio-related topics in DOSBox Staging, so feel free to ask. I implemented many of the audio enhancements in the past few years.

To configure MIDI output, use the `MIXER /LISTMIDI` command to list the available MIDI ports on your system. Then set the following in your config:

```ini
[midi]
mididevice = auto
midiconfig = X
```

Where `X` is either the number of the MIDI port, as listed by `MIXER /LISTMIDI`, or a part of the name of the MIDI port. I always use the first few characters of the name as the numeric ID can be reassigned if you have a lot of USB MIDI devices that you unplug from time to time. E.g. I just set `midiconfig = loopMIDI A` when I use Roland's virtual Sound Canvas running in a separate VST host.

**Reply by madbrain76** [reply]
Very cool. I see in your github that you worked on a MOD player. So did I - almost 3 decades ago.
https://www.os2world.com/wiki/index.php/Dual_Module_Player

I had never heard of the Nim language. I have stuck with C, C++ and some assembly over my career. Just have not been that curious about the other languages out there. First and last interpreted one I use was BASIC.

In case you are also into the analog/acoustic kind of music, take a look at https://www.youtube.com/watch?v=YeDTV8YvwGw .
And to stay on topic, read the comment made by Pyhesty at the bottom of that video . It actually pertains to code I wrote for Silmarils games. I believe the string in question was inside the assembly setup program. I had full permission from the owners of the company. Nowadays hardly anybody writes in assembly, and unused strings tend to be completely removed by optimizing compilers and linkers., not just compress as in the case of the executable. Compression, with things like LZEXE header removal,  was used as a method of preventing patches to remove the copy protection. They were usually no match for about an hour using Turbo debugger 386, and updating the generic TSR I wrote to patch game in RAM after they uncompressed. All that was needed was the address, segment and offset, and the number of 0x90 (NOP) bytes to fill. Or occasionally some JMP for the most complex cases. I once cracked somebody's professional minitel server software right in front of them, after they had spent good money to add copy protection code that depended on a hardware dongle attached to the parallel port. To this day I don't know how the protection was supposed to work, but I didn't care, my patch was completely effective. Those were really the days. Now everything is locked down, signed, hardware protected with stupid DRM, etc. Not to mention horribly bloated with all the abstraction layers in the current operating systems. I believe I coded the audio code for Silmarils on an 8086 machine at just 8 MHz. It could have been a 80286. But definitely not a 80386 or 80486.
I used the pay for my Silmarils code to buy an 80486 with 16MB of RAM, which was huge back then with the awful memory shortages. 
And started developing DOS programs in a VDM, virtual DOS machine, in 1992, under OS/2 2.0. Not everything was emulated as carefully as DOSBox. Sound blaster code sometimes struggled. DMA was problematic. Some games did run, though. Can't remember if Ishar and other Silmarils games did. The VDM came extremely handy if your assembly code under development crashed the system. No reboot or even power off needed, if even CTRL-ALT-DEL stops responding..

**Reply by johnnovak** [reply]
That's some very nice playing skills in the video! Sounds really nice, I love the harpsichord.

I checked out Dual Module Player last year, it runs fine in DOSBox Staging. I was a CapaMod & Fasttracker II guy back in the day, I had a GUS Max.

Nim is quite nice, it's my favourite language for hobby coding. What the MOD player writing exercise taught me was that the only 100% way is to convert the original Amiga 68k assembly playroutine instruction by instruction into C code. Well, I spent many years in ProTracker on the Amiga writing MOD tunes (and listening to them), so I'm very sensitive to some command combinations being slightly off... It's almost impossible to get it 100% right by ear, but I think I got pretty close (although by creating test files and comparing the ProTracker and my tracker's output...)

You might have heard of Audiomonster (Raphaël Gesqua) and Moby (Fredric Motte), they're French too and they're among my favourite MOD composers. Definitely in the top 5!

**Reply by madbrain76** [reply]
Thank you ! I'm very proud of that recording, actually. I dare say it's my best so far. It stood up to scrutiny by world-renowned harpsichordists. Unfortunately it takes me a very long time to learn stuff as I'm a memorizer and not a sight reader. And not getting any younger, so my repertoire isn't growing. I have been mostly absent from the gaming scene since I started learning to play piano & harpsichord in 2001. Had to choose between games and music, and it wasn't a hard choice. But game music - that got me.

Nice to know that DMP runs in DOSbox . I never tried it before - only on bare metal DOS. I can't remember if it ran properly in an OS/2 VDM - I don't believe so. Even if it did, the multi-tasking wasn't that good and the scheduler would break continuity with DOS programs.

When I ported DMP to OS/2, it was a much easier job - at least the same processor family, just 16 bits vs 32 bits. But it turns out much of the assembly code could run with few if any modifications, at least when it came to the mixing part. Everything to do with hardware or actually outputting the sound had to be scrapped. I believe even had to simulate timer interrupts with a separate thread.
The DMP original author made the source public some years ago. I haven't yet put in the effort to post the OS/2 version source. I restored everything from DDS-4 tape 2 years ago. But I have so many versions of the source code in there that I don't know which one is the latest. I didn't have source control of my own back then. Also, while OS/2 will run in Virtualbox, it's not trivial. OS/2 was never trivial, though, not even the first time I went through the 30 5.25" 1.2MB floppies. There is a CD version, but only a few machines ever managed to boot it due to limited driver support. You mostly had to boot with the first two 1.44MB floppies, and then use the CD for the rest. 

I had not heard of Audiomonster, or that Moby. Looked up his credits on Mobygames. Seems like Fascination (1991) is the only game I played that featured his music, and I can't say the music or the game made much of an impression.

I often play the DOS demos on Youtube for their music only. In fact, I play the music in Firefox on my Android phone, then cast the device audio to one of the 20+ streamers in my home, and it ends up on on a couple or some combination of the 70+ speakers, indoors or outdoors. Typically, it's just 2 speakers at a time, though.

**Reply by johnnovak** [reply]
Yeah, the Fascination soundtrack is nothing special. Audiomonster and Moby are mostly famous for their Amiga demo tracker tunes. Here are some of my favourites. I studies these a lot back in the day, they use tracker commands and samples masterfully. All of these are between 50 and 200 KB in size, so you had to be very economical with samples.

But then, they might sound nothing special in 2024, but to me (and many others) they were magic to create such music that fits on 20-30% of a floppy of a 1985 computer 😄 

https://www.youtube.com/watch?v=1wvyZ-WWQtI

https://www.youtube.com/watch?v=7jraaxcuXLk

https://www.youtube.com/watch?v=zAdMBc3PgXw

https://www.youtube.com/watch?v=rpoNUubVyD8&pp=ygULbW9ieSBtb2J5bGU%3D

https://www.youtube.com/watch?v=XgIy9_mJDqQ&pp=ygUUYXVkaW9tb25zdGVyIHRoZSB3YXI%3D

https://www.youtube.com/watch?v=8RJgPNOyW14

https://www.youtube.com/watch?v=khSTGdeKhIA

**Reply by madbrain76** [reply]
I listsened to all those. We might have slightly different taste in music. I think there is a strong nostalgia factor involved. I like Purple Motion and Lizardking. When I first moved to the US, there was another OS/2 programmer in the same town, by coincidence. We collaborated on porting a DOS demo to OS/2 I. believe it's the only one that was ever released. It featured a MOD file from Purple Motion.

https://www.youtube.com/watch?v=GvIjup2PJiY

**Reply by johnnovak** [reply]
Sure, there is no arguing with taste 😄 Audiomonster are Moby are more like jazz and rock guys. I used to study classical guitar, and I like classical music too, but I remember there was a sharp divide between most guys on the classical vs jazz path in the music school.

Purple Motion is nice too, he's clearly more classically influenced. I like his tunes too. I grew up with C64 music, it's still one of my favourite forms of computer music.

**Reply by madbrain76** [reply]
My father kept me from getting those early 8-bit computers. So the first one I got was a PC with a 16-bit 8086, with 4 color CGA and PC speaker. Let's just say games weren't too great on that. There was a special 640x200x4 bit mode on that graphics card. It ran GEM. But almost no games used it. I was just on a thread on YouTube with Stephane Picqu talking about Captain Blood being so awful in the PC port. But Commander Blood was great. He says he's going to remaster it like he did for Dune. Dune was great on MT-32 growing up.

**Reply by johnnovak** [reply]
Yeah, my progression was C64 -> Amiga 500 (spent a lot of time in ProTracker writing MOD tunes), then 486DX2/66 in 1995 with a Gravis Ultrasound.

Captain Blood was quite nice on the C64 and the Amiga. It was an Atari ST original.

Stéphane Picq is one of my favourite composers ever, the Dune soundtrack is so great! IMO, it's best on the AdLib Gold with the additional Surround Module. I've added emulation of the Surround Module to Staging specifically for Dune, as no other game uses it. But it was worth it! 😎 

https://www.dosbox-staging.org/releases/release-notes/0.79.0/#adlib-gold-surround-module-emulation

Real hardware recording:
https://www.youtube.com/playlist?list=PLDE06DE88BB135F33

I'm quite fond of French games in general (but I don't speak French) 😄 Delphine, Coktel Vision, Cryo Interactive, most of their games are so unique and different compared to their American counterparts.

**Reply by madbrain76** [reply]
I have had so much hardware in my life it's impossible to remember, let alone list it all. But I know I started with an 8086, then 80286, than 80386 SX, 80486DX, 80486 DX2, Pentium 100, dual CPU Pentium 100, K6/3 550, Dual Athlon MP, Core 2 Q6600, 5820k, AMD 5950X, i5-6600k, i5-6600, i5-12400 i5-13400, AMD 5700G,. That's a lot of machines, and I built them all myself except the first two. The reason there are so many CPUs is that I frequently updated the motherboard without the rest of the parts. Currently have 6 machines in the house. One is my silent triple-WAN 10Gbe router, another is a 20TB flash NAS, also silent, 2 are HTPCs, one is a 112TB HDD NAS.
I built two new machines last year, but reusing many old parts. Upgraded one the year before that. It's a rare year that none of the 6 PCs gets upgraded in some fashion (used to be just 4 total).

Audio / MIDI wise I remember owning AdLib first, then Soundblaster 1.x , SB Pro, SB 16., Pro audio Spectrum 16, LAPC-1. MIDI Quest MQX-16, CM-32L,  MT-32, FP-3, SC-88VL, Fantom XR, UM-880, VK-88. There is probably a lot of stuff I'm forgetting. And if I started listing the amplifiers, receivers, speakers, etc, I would never get to the end of it. The current 70 speakers I mentioned aren't an exaggeration at all, it's very close to the real count, which I did in a spreadsheet once. 32 of them came with the house, built-in to walls, ceilings, even outdoors. But 14 years ago we didn't have streamers to use them remotely. A CD changer with an RF remote was it. 

I have a garage full of old computer parts and cables. You never know when you're going to need that gold-plated SCART cable for a French TV. At some point I had an import Sony CRT that could play SECAM. and used SCART .. But for the last 21 years my HT uses projectors - I went through 4, but still the same 106" screen. We do have a 36" 480i Sony CRT in our bar. It's there because it exactly fills the space of the built-in cabinet. It would look weird without it. We got it free off craigslist. We barely use it. I tried to hook up Raspberry Pi 3B+ with it using the composite output. There are major overscan problems in the video drivers, bad enough that you can only see about 60% of the intended image, and it's completely unusable. But my husband's LaserDisc player still works fine on that CRT without any issue. As does a Samsung DVD player. The overscan problems are definitely with the Pi drivers.

The flash NAS also runs a Home Assistant VM. That works with the 12 WiiM Pro plus streamers, and the many Chromecast and Fire Stick 4K video sticks in the various rooms. I can stream audio from the flash NAS. Or from Qobuz. There are 170 DHCP deservations on my network, mostly IoT devices. Another 43 Z-Wave, 15 Yolink. It's a bit of a time sink to manage all this. Oh, I forgot the 6 Wifi APs, 4 of them meshed, since there is no Ethernet in the house, and it can't be added inside, sigh.

Right now in my home office I have about 150 spare parts I don't really know what to do with. They are probably worth nothing, but I don't really want to trash/recycle them. So, they just sit there on the floor for cats to play with.

**Reply by madbrain76** [reply]
I did not know there was a surround sound module for the AdLib Gold. Was it using Dolby Prologic ? Youtube only supports 2 channels. DSMI/DMP supported Prologic. And I had a 4 channel receiver setup at one point to make use of it.

I still think the Dune music sounds better on MT-32, even though I know it wasn't composed for it. Kind of the way Bach sounds good on nearly any instrument., in competent hands, never mind what historically informed performers will say - and there many in the harpsichord world. You have to intentionally choose an instrument inappropriate for a Bach piece it not to work. There are transcriptions I dislike, like, say, the Goldberg Variations on solo marimba. But apparently some people like them. But I won't buy a disc on that instrument again.

I had both an SB16 and LAPC-1 at the time I first played Dune. Actually, I still have an ISA bus SB16 in a drawer. I have no motherboard left to plug it into. Comparing the audio/music between cards was always interesting. I don't think I recall one game that for which I preferred the FM version. I didn't bother to compare the PC speaker version. though. I'm not a masochist.
 :
Since we were talking about MODs, this came up on Youtube after listening to other modules :
https://www.youtube.com/watch?v=thnXzUFJnfQ
That was definitely one of my favorites. And I probably have a copy on those DAT/DDS-4 I restored 4 years ago, booting OS/2 on bare metal with an ancient Adaptec Ultra320 SCSI card to hook up the HP tape drive, and some obscure VHDCI SCSI cable. The non-Y2K compliant BakupWiz program didn't preserve the original file dates during restore, unfortunately. I still have a drawer full of SCSI cables. I'm ready for the next retro storage trend ! Along with a still working (USB) floppy drive and a bunch of floppies that still format without too many bad sectors. I still preserved some non-working internal floppy drives, even though the floppy connector has been gone from motherboards for some time, before even IDE disappeared. Got another drawer for SATA/IDE cables too.

Anyway, I have that Space Debris MOD file, and a few others, on that 112TB volume.

```
X:\Tapes\Charm\Musique>dir/S space*
 Volume in drive X is ZFS```

 Volume Serial Number is 509A-562D

 Directory of X:\Tapes\Charm\Musique\Mod\mod

03/21/2020  14:26           347,582 SPACEDEB.MOD
```

Dope was another, from a demo. It is on Youtube.

```
X:\Tapes\Charm>dir/S dope*
 Volume in drive X is ZFS
 Volume Serial Number is 509A-562D

 Directory of X:\Tapes\Charm\DEV\PROJETS\DSMI3\Programs\OSCILLOSCOPE

03/21/2020  14:24            99,938 dope.mod

```

PANIC.S3M was cool too - another demo.

As to French games, I agree they have been pretty creative in general. But when it came to PC releases, they were not at the forefront, let's just say. I think 3 to 5 years behind the US companies in that area. And that goes for graphics too. Way too many games released only in CGA versions. Like Skweek. I never knew they later made an EGA version. There was no way to get updates those days, other than "floppies in the mail in a package with, shall we say, 'recycled' stamps" , and since the update wasn't publicized, no one knew to bother looking for it. Also, the Skweek opening screen has decent but short sampled PCM music, for PC speaker. But afterwards it's all insufferable beeps.

I believe I got my AdLib card in 1989, when I still had an 8086. There was basically nothing in France that supported it for years. Same for Roland. It took many years for publishers to catch up.  Whereas companies like Lucasfilm and Sierra were really leading the pack. But it was certainly frustrating to play games in English, while I was studying German as my first foreign language, and English years later. Try understanding a print version of an AdLib SDK in English not knowing the language. Which is sad of AdLib, since they were a Canadian out of Québec, but didn't publish the SDK in French.

I got the LAPC-1 later. There an extremely annoying port conflict between the Adaptec 1542 SCSI card and the MPU-401 MIDI card (part of the LAPC-1). They both used port 330h by default.  Both had jumpers to change to alternate ports, like 332h and 334h .
But games would usually only look for the MIDI card at port 330h, not the other ports.
And the Adaptec BIOS also only supported port 330. That meant that if you changed the port on it, your SCSI drive was no longer bootable. And of course I was booting from SCSI, and had games using MIDI on the hard drive.
So, I dumped the Adaptec ROM to a file and fixed all the instances of 0330 with a hex editor. Then my father took the file to his lab where he had an EEPROM programer, and I got it back the next day. This is back when those chips were socketed. Anyway, it didn't work. Then, I figured out there was checksum built in to the RAM. It had to add up to all zeroes. I changed the last couple of bytes to make that happen. My father burnt the EEPROM again. And my Adaptec SCSI card and MIDI card lived happily ever after, together.

Anyway, I hope we get to the bottom of that Silmarils sound issue. Might be worth a try in Virtualbox to see how it behaves there. Pretty sure I can still manage to boot DOS bare metal on several PCs, but I don't have an AdLib/SB hardware compatible that can plug into a PCI or PCI-E slot. Apparently, some exist in PCI format. One of my 6 machines still has 1 legacy PCI slot. The games might run too fast on bare metal. You can however write a TSR that uses timer interrupt to run busy loops and consume any extra amount CPU cycles . I remember there were such programs to slow things down for games that were too fast on a 12 MHz 286. Mine had a hardware "Turbo" button that could drop it down to 8 MHz...

**Comment by AnonTet** [comment]
This thread is already worth it.
I've grown listening to electronic music and this sounds just as good today as it did then.
Getting a bit of history on top of this is just the cherry on top.

**Comment by madbrain76** [comment]
Actually, I just located my 1992 sources for the sound in those games, (those tapes really did come in handy) as well as the setup. Just not sure my code is being used for the initial sound in those Silmarils games. Definitely not in anything pre-1992, but starting with Ishar, possibly. It would be helpful if the Dosbox log could tell at what point a device starts playing, even if its just the PC speaker. Maybe it logs this already.. I'm fairly new to DOSbox.

**Reply by johnnovak** [reply]
Nice find! Yeah I will look into this again in the coming days and report back.

**Comment by johnnovak** [comment]
@AnonTet @madbrain76 I'm happy to announce the issue has been resolved in https://github.com/dosbox-staging/dosbox-staging/pull/3772 🎉
